File: app/src/modules/app/api/settings/data-transfer-objects/create-service.ts
import { ServiceConfiguration } from '@/modules/app/api/settings/entities/service-configuration';
import { VehicleCategory } from '@/modules/app/api';
import { ServicePricingMethod } from '@/modules/app/api/settings/value-objects/service-pricing-method';

export type CreateService = {
    name: string;
    description?: string;
    pricingMethod: ServicePricingMethod;
    configurations: { vehicleCategoryId: string | null; price: number; duration: number }[];
};

File: api/src/Domain/Settings/DataTransferObjects/CreateService.cs
using System;
using System.Collections.Generic;

namespace DetailingArsenal.Domain.Settings {
    public class CreateService : IDataTransferObject {
        public Guid UserId { get; }
        public string Name { get; }
        public string? Description { get; }
        public ServicePricingMethod PricingMethod { get; }
        public List<CreateServiceConfiguration> Configurations { get; }

        public CreateService(Guid userId, string name, string? description, ServicePricingMethod pricingMethod, List<CreateServiceConfiguration> configurations) {
            UserId = userId;
            Name = name;
            Description = description;
            PricingMethod = pricingMethod;
            Configurations = configurations;
        }
    }
}